esp/public/media/scripts/onsite/ajax_status.js

```diff
@@ -1293,15 +1294,19 @@ function populate_counts()
     {
         var sec_id = data.counts[i][0];
         var num_students = data.counts[i][1];
+        var num_students_attending = data.counts[i][2];
         
         //  If we have a conflict, assume the larger number of students are enrolled.
         if (!data.sections[sec_id])
             console.log("Could not find section " + sec_id);
-        else if (data.sections[sec_id].num_students_enrolled != num_students)
-        {
-            //  console.log("Warning: Section " + sec_id + " claims to have " + num_students + " students but " + data.sections[sec_id].num_students_enrolled + " are enrolled.");
-            if (num_students > data.sections[sec_id].num_students_enrolled)
-                data.sections[sec_id].num_students_enrolled = num_students;
+        else {
+            data.sections[sec_id].num_students_attending = num_students_attending;
+            if (data.sections[sec_id].num_students_enrolled != num_students)
+            {
+                //  console.log("Warning: Section " + sec_id + " claims to have " + num_students + " students but " + data.sections[sec_id].num_students_enrolled + " are enrolled.");
+                if (num_students > data.sections[sec_id].num_students_enrolled)
+                    data.sections[sec_id].num_students_enrolled = num_students;
+            }
```

Comment:
I know you didn't write this part, but I'm a little confused about what's going on here. Is this saying that we only update the enrollment numbers if they are _higher_ than they were before? Shouldn't we just always update them like you've done with `num_students_attending`?

Reply:
So num_students is the derived field of enrollment for the section whereas data.sections[].num_students_enrolled originally comes incrementing the enrollment of the section one by one as they loop through student enrollments in line 1257 (populate enrollments). So if they don't match they take the higher of the two. For students checked in to the program, they only do the looping, not the derived field, whereas for class attendance, I only do the derived field because it seems fastest & easiest. Perhaps this code needs to be cleaned up in the future...